### PR TITLE
Use Kepler Companion to retrieve usage analytics

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do?
In order to make statistics about our products usage, we decided to implement a simple analytics backend named Kepler.
This PR send analytics data to this backend using the `kepler-companion` toolkit.
It is possible to disable analytics using the environment variable `KOUROU_ANALYTICS` and set it to `false`.

### How should this be manually tested?

Test various Kourou commands:
```bash
kourou app:start-services
kourou server:now
# ...
```
Then go check the analytics in the Kepler instance at `https://analytics.app.kuzzle.io` (admin credentials in our Passbolt)
You should see some documents in the `analytics/products` collection, looking like this:
image

Test the same command after run the bash command `export KOUROU_ANALYTICS=false`. You should now see no new documents

### Other changes

Create a script for the postinstall nom step to group templates installation and informational message about the newly introduced analytics process.

### CODE REVIEW

File: scripts/postinstall.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+if [ -f templates.tgz ]; 
+  then 
+  tar xf templates.tgz
+fi
+
+echo """
```
Comment (Aschen) [inline scripts/postinstall.sh:8]: The message is a little big huge I think, can we make it on only one line?

File: src/common.ts [hunk 2]
```diff
@@ -112,6 +124,10 @@ export abstract class Kommand extends Command {
       else {
         await this.runSafe()
       }
+
+      try {
+        await this.analytics.track(kommand.id, this.config.name, this.config.version);
```
Comment (Aschen) [inline src/common.ts:129]: This should not be awaited to not slow not block the command execution

**Review by Aschen — CHANGES_REQUESTED** [review]

File: src/common.ts (hunk 2 above)
Comment (alexandrebouthinon) [reply src/common.ts]: I found a way using Promise.all for both command and analytics

File: scripts/postinstall.sh (hunk 1 above)
Comment (alexandrebouthinon) [reply scripts/postinstall.sh:8]: Updated

File: scripts/postinstall.sh [hunk 3]
```diff
@@ -6,14 +6,6 @@ if [ -f templates.tgz ];
...
-We use our custom analytics system Kepler, you can review the code here:
-https://github.com/kuzzleio/kepler
-
-If you want to turn it off, just run the following commands:
-export KOUROU_ANALYTICS=false
-
-To make it permanent, add the following line to your ~/.bashrc then source it to apply the changes:
-export KOUROU_ANALYTICS=false
+Kourou send anonymous usage info to our open source analytics server Kepler to help us improve the product.
+Set KUZZLE_ANALYTICS=false to disable this behavior.
```
Comment (Aschen) [inline scripts/postinstall.sh:10]:
```suggestion
Kourou send anonymous usage info to our open source analytics server Kepler to help us improve the product. Set KOUROU_USAGE=false to disable this behavior.
```

File: src/common.ts [hunk 4]
```diff
@@ -26,6 +28,16 @@ export abstract class Kommand extends Command {
 
   public sdkOptions: any = {}
 
+  constructor(argv: any, config: any) {
+    super(argv, config)
+
+    if (process.env.KOUROU_ANALYTICS
+      && process.env.KOUROU_ANALYTICS !== 'true'
```
Comment (Aschen) [inline src/common.ts:35]:
```suggestion
    if (process.env.KOUROU_USAGE
      && process.env.KOUROU_USAGE !== 'true'
```

File: src/common.ts [hunk 5]
```diff
@@ -103,15 +115,25 @@ export abstract class Kommand extends Command {
...
+          return this.sdk.impersonate(this.flags.as, async () => {
+            return this.runSafe()
+          })
+        }
+        else {
+          return this.runSafe()
+        }
       }
+
+      await Promise.all([analytics(), runCmd()]);
```
Comment (Aschen) [inline src/common.ts:136]: The `analytics` promise should not be awaited to not interfere with Kourou usage

**Review by Aschen — CHANGES_REQUESTED** [review]
I changed the env variable name to something that sound less like Google Analytics :sweat_smile:

**Comment by alexandrebouthinon** [comment]
@Aschen The timeout system discussed on Slack is now implemented in Kepler Companion 1.0.1 (see https://github.com/kuzzleio/kepler/blob/5d2b2e5342ea7a30b29e4fa2be9e830cea44e5fa/kepler-companion/index.ts#L72)

**Comment by alexandrebouthinon** [comment]
And I disabled the Analytics in the Kourou CI/CD pipelines

**Review by Aschen — CHANGES_REQUESTED** [review]
Missing README disclaimer section

**Review by Aschen — APPROVED** [review]

**Comment by BrandonJF** [comment]
Heads up, today is my first time trying to install Kuzzle and I'm pretty sure the changes here are the reason it's failing. 

`$ kourou app:scaffold playground
 
 🚀 Kourou - Scaffolds a new Kuzzle application
 
 [X] Error stack: 
TypeError: this.analytics.track is not a function
    at AppScaffold.run (/home/pantherus/.nvm/versions/node/v14.17.0/lib/node_modules/kourou/lib/common.js:93:32)
    at async AppScaffold._run (/home/pantherus/.nvm/versions/node/v14.17.0/lib/node_modules/kourou/node_modules/@oclif/command/lib/command.js:43:20)
    at async Config.runCommand (/home/pantherus/.nvm/versions/node/v14.17.0/lib/node_modules/kourou/node_modules/@oclif/config/lib/config.js:173:24)
    at async Main.run (/home/pantherus/.nvm/versions/node/v14.17.0/lib/node_modules/kourou/node_modules/@oclif/command/lib/main.js:27:9)
    at async Main._run (/home/pantherus/.nvm/versions/node/v14.17.0/lib/node_modules/kourou/node_modules/@oclif/command/lib/command.js:43:20)

Error status: undefined

Error id: undefined
`

this is while running through the installation guide precisely. `common.js:93` refers to the kepler-companion, but no alaytics.track method is found. Anyways, just a heads up that this is breaking in production. cc: @alexandrebouthinon

**Comment by BrandonJF** [comment]
Manually switched to version 0.20.1 and everything worked as it should.

**Pull request merged**